src/main/java/edu/harvard/iq/dataverse/engine/command/impl/UpdateDataverseInputLevelsCommand.java

```diff
@@ -0,0 +1,50 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.DataverseFieldTypeInputLevel;
+import edu.harvard.iq.dataverse.MetadataBlock;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.engine.command.AbstractCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredPermissions(Permission.EditDataverse)
+public class UpdateDataverseInputLevelsCommand extends AbstractCommand<Dataverse> {
+    private final Dataverse dataverse;
+    private final List<DataverseFieldTypeInputLevel> inputLevelList;
+
+    public UpdateDataverseInputLevelsCommand(Dataverse dataverse, DataverseRequest request, List<DataverseFieldTypeInputLevel> inputLevelList) {
+        super(request, dataverse);
+        this.dataverse = dataverse;
+        this.inputLevelList = new ArrayList<>(inputLevelList);
+    }
+
+    @Override
+    public Dataverse execute(CommandContext ctxt) throws CommandException {
+        if (inputLevelList == null) {
+            throw new CommandException("Exception while updating dataverse input levels: Input level list cannot be null", this);
+        }
+        addInputLevelMetadataBlocks();
+        return ctxt.engine().submit(new UpdateDataverseCommand(dataverse, null, null, getRequest(), inputLevelList));
```

Comment:
Also you need to set isMetadataBlockRoot to true so that the Dataverse knows that it not inheriting its MetadataBlocks from its parent - that is it has a different input level, etc.

Reply:
I just added that line and also check that the input list is not empty